src/UNL/UCBCN/ActiveRecord/Record.php

```diff
@@ -317,6 +326,39 @@ public static function getByAnyField($class, $field, $value, $whereAdd = '')
         return $record;
     }
 
+    public static function getManyByAnyField($class, $field, $value, $whereAdd = '') {
+        $records = array();
+        $record = new $class;
+
+        if (!empty($whereAdd)) {
+            $whereAdd = $whereAdd . ' AND ';
+        }
+
+        $mysqli = self::getDB();
+        $sql    = 'SELECT * FROM '
+                    . $record->getTable()
+                    . ' WHERE '
+                    . $whereAdd
+                    . $field . ' = "' . $mysqli->escape_string($value) . '"';
+        $result = $mysqli->query($sql);
+
+        if (false === $result) {
+            throw new Exception($mysqli->errno.':'.$mysqli->error, 500);
+        }
+
+        if ($result->num_rows == 0) {
+            return false;
+        }
+
+        while (!empty($row = $result->fetch_assoc())) {
+            $record = new $class;
```

Comment:
That's fair. Since we discussed that the N+1 query problem was not an issue the other day, I can change this up, and remove the method I created. Some more advanced ORM's do have ways of basically doing it either way, yes. I think Sequel (a Ruby gem) all you have to do to switch it is put .lazy in the middle of your query method string and it will return an iterator instead of an array.